packages/graphql/src/translate/connection/create-connection-and-params.ts

```diff
@@ -52,7 +52,18 @@ function createConnectionAndParams({
 
     const subquery = ["CALL {", `WITH ${nodeVariable}`];
 
-    const sortInput = resolveTree.args.sort as ConnectionSortArg[];
+    const sortInput = (resolveTree.args.sort ?? []) as ConnectionSortArg[];
+    // Fields of {edge, node} to sort on. A simple resolve tree will be added if not in selection set
+    // Since nodes of abstract types and edges are constructed sort will not work if field is not in selection set
+    const sortFields: Record<keyof ConnectionSortArg, string[]> = Object.assign(
+        {},
+        ...(["edge", "node"] as Array<keyof ConnectionSortArg>).map((element) => ({
+            [element]: ([] as string[]).concat(
+                ...sortInput.map(({ [element]: sortField = {} }) => Object.keys(sortField))
+            ),
+        }))
+    );
```

Comment:
I'm honestly having a hard time understanding what this is actually doing so can't make a suggestion as to how it could be a bit more readable... But the `assign` -> `map` -> `concat` -> `map` makes this pretty hard to comprehend. Any chance it could be simplified a touch - even if it's not quite as concise!!!

Reply:
I think this one got away from me.
Refactored in https://github.com/neo4j/graphql/pull/646/commits/fcfd3091089cec432f34064eea2e61ccaadc33bd